fe/fe-core/src/main/java/org/apache/doris/datasource/jdbc/client/JdbcMySQLClient.java

```diff
@@ -374,11 +377,11 @@ private Type dorisTypeToDoris(String type) {
                 return ScalarType.createDatetimeV2Type(scale);
             }
             case "CHAR":
-            case "VARCHAR": {
-                int length = Integer.parseInt(upperType.substring(openParen + 1, upperType.length() - 1));
-                return baseType.equals("CHAR")
-                    ? ScalarType.createCharType(length) : ScalarType.createVarcharType(length);
-            }
+                ScalarType charType = ScalarType.createType(PrimitiveType.CHAR);
+                charType.setLength(fieldSchema.getColumnSize().orElse(0));
+                return charType;
+            case "VARCHAR":
+                return ScalarType.createVarcharType(fieldSchema.getColumnSize().orElse(0));
```

Comment:
Curious to know why do we set length to 0 when column size is not set?

Reply:
Thanks for finding this problem. The processing here is not appropriate, so I changed the behavior here, refer to https://github.com/apache/doris/pull/41243